plugins/org.eclipse.papyrus.uml.diagram.sequence.runtime/src/org/eclipse/papyrus/uml/diagram/sequence/runtime/internal/edit/policies/AbstractSequenceGraphicalNodeEditPolicy.java

```diff
@@ -245,6 +246,11 @@ protected FeedbackHelper getFeedbackHelper(CreateConnectionRequest request) {
 	protected Command getReconnectSourceCommand(ReconnectRequest request) {
 		Command result = super.getReconnectSourceCommand(request);
 
+		if (isForce(request)) {
+			// Just reconnect this end; no sloping constraints required
+			return result;
```

Comment:
When grabbing a message from the center, the request is forced. Thus we do not check validity of the move using _constrainReconnection_. In this case, we can grab and reorder graphically messages, while their ends are semantically not reordered. Is this done on purpose?

Reply:
The constraints referenced here are the constraints on how a message may be sloped:  not at all for synchronous messages and forwards in time only for asynchronous messages.

The re-ordering constraint is implemented in the calculation of the semantic reconnection command in the graphical node edit-policy.  So, the feedback should show the connection moving but then the mouse pointer may show the no-smoking cursor.

Reply:
I do not get this no-smoking cursor. I can currently shift a sloped message, so the end of this message is graphically placed after the following message for example.  Debugging lead me to that place, I was expecting to go though method constrainReconnection(ReconnectRequest request, MMessageEnd end) defined in commit #7003616, but this is short circuited by the force parameter on the request

Reply:
Ah, right.  Sorry, I thought we were looking at the feed-back helper here.  I'll look into it.